src/components/ModalHeader.tsx

```diff
@@ -0,0 +1,57 @@
+import { Ref, forwardRef } from 'react'
+import { Flex, FlexProps, P } from 'honorable'
+import PropTypes from 'prop-types'
+
+import CloseIcon from './icons/CloseIcon'
+
+type ModalHeaderProps = FlexProps & {
+  title?: string
+  onClose?: () => void
+}
+
+const propTypes = {
+  title: PropTypes.string,
+  onClose: PropTypes.func,
+}
+
+function ModalHeaderRef({ children, onClose, ...props }: ModalHeaderProps, ref: Ref<any>) {
+  return (
+    <Flex
+      ref={ref}
+      align="center"
+      justify="space-between"
+      marginBottom="large"
+      {...props}
+    >
+      <P
+        overline
+        color="text-xlight"
+      >
+        {children}
+      </P>
+      {typeof onClose === 'function' && (
+        <Flex
+          align="center"
+          justify="center"
+          padding="small"
+          margin={-12}
+          borderRadius="50%"
```

Comment:
I'm pretty sure we shouldn't round this border actually, probably use something like a 2-3px radius.  We can confirm with nick but it looks clunky